src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Binders/PlaceholderLocalBinder.vb

```diff
@@ -43,13 +44,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
                 Return
             End If
 
+            Dim canonicalName = Canonicalize(name)
+
             Dim local As LocalSymbol = Nothing
             If _implicitDeclarations IsNot Nothing Then
-                _implicitDeclarations.TryGetValue(name, local)
+                If Not _implicitDeclarations.TryGetValue(canonicalName, local) Then
```

Comment:
I thought that the value was undefined when false was returned, but apparently it's actually spec'd to be the default value of the type.